onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -220,26 +268,77 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
                 return E_UNEXPECTED;
             }
 
-            if (outputTensor->GetTensorDataType() != MLOperatorTensorDataType::Float ||
-                inputTensor->GetTensorDataType() != MLOperatorTensorDataType::Float)
-            {
-                return E_UNEXPECTED;
-            }
-
             ComPtr<IUnknown> executionObject;
             ComPtr<ID3D12GraphicsCommandList> commandList;
             context->GetExecutionInterface(executionObject.GetAddressOf());
             executionObject.As(&commandList);
 
-            auto dftParams = PrepareDFT(context);
+            // Get the input and output shape sizes
+            auto inputDims = GetTensorDimensions(inputTensor.Get());
+            ML_CHECK_VALID_ARGUMENT(static_cast<size_t>(m_axis) < inputDims.size())
+            auto outputDims = GetTensorDimensions(outputTensor.Get());
+            ORT_THROW_HR_IF(E_FAIL, inputDims.size() != outputDims.size());
+
+            ComPtr<IUnknown> inputUnknown;
+            ComPtr<ID3D12Resource> inputResource;
+            inputTensor->GetDataInterface(inputUnknown.GetAddressOf());
+            inputUnknown.As(&inputResource);
+
+            ComPtr<IUnknown> outputUnknown;
+            ComPtr<ID3D12Resource> outputResource;
+            outputTensor->GetDataInterface(outputUnknown.GetAddressOf());
+            outputUnknown.As(&outputResource);
+
+            // Get optional dft_length input
+            uint32_t dftLength = inputDims[m_axis];
+            ComPtr<IMLOperatorTensor> dftLengthTensor;
+            if (SUCCEEDED(context->GetInputTensor(1, &dftLengthTensor)) && dftLengthTensor != nullptr)
```

Comment:
> && dftLengthTensor != nullptr [](http://example.com/codeflow?start=71&length=30)

Redundant. COM out param contract is that a function returns a non-nullptr if and only if succeeded.

Reply:
Unfortunately the SAL annotation and comments are in disagreement, and the original code relied on it succeeding even if it writes nullptr.

```cpp
    //! Gets the input tensor of the operator at the specified index.
    //! This sets tensor to nullptr for optional inputs which do not exist.
    //! Returns an error if the input at the specified index is not a tensor.
    STDMETHOD(GetInputTensor)(
        uint32_t inputIndex,
        _COM_Outptr_result_maybenull_ IMLOperatorTensor** tensor
        ) const noexcept PURE;
```